presto-geospatial/src/main/java/com/facebook/presto/geospatial/GeoFunctions.java

```diff
@@ -561,9 +561,14 @@ public static Boolean stWithin(@SqlType(GEOMETRY_TYPE_NAME) Slice left, @SqlType
 
     private static OGCGeometry geometryFromText(Slice input)
     {
-        OGCGeometry geometry = OGCGeometry.fromText(input.toStringUtf8());
-        geometry.setSpatialReference(null);
-        return geometry;
+        try {
```

Comment:
we can only wrap the code that can throw the `IllegalArgumentException`:

```java
OGCGeometry geometry;
try {
    geometry = OGCGeometry.fromText(input.toStringUtf8());
}
catch (IllegalArgumentException e) {
    throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Invalid WKT: " + input.toStringUtf8(), e);
}
geometry.setSpatialReference(null);
return geometry;
```

(Note that this also changed the `throw` statement).